### PR TITLE
Add question service

### LINKED ISSUE #30: FR3.1
The system should be able to store question banks indexed by difficulty level.

### PR DESCRIPTION
**Pull request by tanwayne890**
Closes #30 #31 

This PR adds:
- APIs to create question and fetch a question randomly based on difficulty level. 
- show a question in `frontend` for both matched user. 
In `matching-service`, when a match is found, it will fetch a question from `question-service` and then send the questionId to both users through `matching-service` socket. `frontend` will fetch the question by the id for each matched user.
- fix `npm run lint:fix`

Case when the question has multiple examples will look ugly since we store a string for the whole question. As discussed, to use 
 an array of strings to separate to be implemented in next PR:
image

### CODE REVIEW

File: frontend/src/components/room/RoomPage.js [hunk 1]
```diff
@@ -125,14 +140,17 @@ function RoomPage({ socket }) {
...
                   </Typography>
                 </Paper>
               </Grid>
             </Grid>
           </Box>
           <Paper variant="outlined" square>
-            <Typography sx={{ height: "40rem" }}>Question</Typography>
+            <Typography sx={{ height: "1.5rem", textDecoration: "underline" }}>
+              {question.title}
+            </Typography>
```
Comment (TianYong-Goh) [inline frontend/src/components/room/RoomPage.js:152]: Screenshot 2022-10-09 at 12 05 28 AM

I have update a new design. the title should be at the top of the box as shown in the picture. I will update it in my pr maybe you could remove this 3 lines from your pr. Thank you

**Review by oliviajohansen — CHANGES_REQUESTED** [review]
matching not working on your branch but working on main?

https://user-images.githubusercontent.com/55739990/194758154-2daf69c8-bd6f-4b2c-a5c9-732f47a897ba.mov

File: frontend/src/components/room/RoomPage.js (hunk 1 above)
Comment (tanwayne890) [reply frontend/src/components/room/RoomPage.js]: https://github.com/CS3219-AY2223S1/cs3219-project-ay2223s1-g23/pull/34/commits/63f0b4ca306d862d3a6d2c80a511896d64a5e06e

File: matching-service/src/controllers/socketController.js [hunk 2]
```diff
@@ -50,6 +66,7 @@ export const initSocketEventHandlers = (socket, io) => {
         socketId2: socketIdFromQueue,
         roomId: roomId,
         difficulty: difficulty,
+        ques: ques
```
[the remark this answers is not in the document]
Comment (tanwayne890) [reply matching-service/src/controllers/socketController.js]: https://github.com/CS3219-AY2223S1/cs3219-project-ay2223s1-g23/pull/34/commits/61e89babb61eb8ba703600726811a49e1051a2cf

**Comment by oliviajohansen** [comment]
didnt set env file so matching wasnt working

**Review by oliviajohansen — APPROVED** [review]
LGTM

**Pull request merged**